### PR TITLE
CI: Change trigger event for docs workflow, add filter

### PR DESCRIPTION
**Pull request by errordeveloper**

### CODE REVIEW

**Comment by errordeveloper** [comment]
@pchaigno you were right in your comment earlier!

**Review by aanm — COMMENTED** [review]
Will this work if the developer push force into the PR?

**Review by pchaigno — APPROVED** [review]
:+1: for the `path` filter!

> Will this work if the developer push force into the PR?

@aanm [The documentation](https://developer.github.com/v3/activity/events/types/#pullrequestevent) is not super clear on this, but it doesn't seem to distinguish between different types of `pull_request/synchronize` events, so we should be fine. I also don't remember seeing any such issues at [github/linguist](https://github.com/github/linguist) and we rely mostly on the `pull_request` event, without any special config.

@errordeveloper It could be good to keep the `pull` event, e.g. if we want to check the test results before actually opening the pull request.

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: Documentation Updates
 
-on: [push]
+on:
+  push:
+    paths: ["Documentation/**"]
+  pull_request:
+    paths: ["Documentation/**"]
```
Comment (pchaigno) [inline .github/workflows/docs.yaml:7]: @errordeveloper Do you know if the following works?
```yaml
- [push,pull_request]:
    paths: ["Documentation/**"]
```

Comment (errordeveloper) [reply .github/workflows/docs.yaml:7]: I'm pretty sure it won't. The `[push,pull_request]` would be too weird as a key name. I was actually hoping that the following may work:
```
  - push:
      paths: ["Documentation/**"]
  - pull_request:
      paths: ["Documentation/**"]
```
But it didn't. Personally, I find ambiguity of config files like this and magic shortcut very annoying, how much time people have to spend guessing! The only sensible answer would be to ditch JSON/YAML and use a scripting language, but then GitHub would have to support at least a few different languages...

**Comment by errordeveloper** [comment]
@aanm I believe that a force-push is just a push as far as GitHub events are concerned.
Also, from my experience so far, if a job was triggered before a force-push occurred it won't get cancelled automatically, but it would fail to e.g. checkout the non-existing commit, if it does tries to do a checkout.

**Pull request merged**